### PR TITLE
Fixed for AR52

### PR DESCRIPTION
**Pull request by rsov**
- Log in rails dev server console now shows where query was executed from
- `User.exists? 1` won't fail due to ID not being passed

### CODE REVIEW

File: lib/arjdbc/abstract/core.rb [hunk 1]
```diff
@@ -74,3 +74,18 @@ def log(sql, name = "SQL", binds = [], type_casted_binds = [], statement_name =
...
 end
+
+
+module ActiveRecord
+  class LogSubscriber
+
+    JDBC_GEM_ROOT = File.expand_path("../../../..", __FILE__) + "/"
+
+    # Remove this gem from log trace so that query shows where it was called in application
+    def ignored_callstack(path)
```
Comment (rdubya) [inline lib/arjdbc/abstract/core.rb]: I'm not sure I follow what this is doing 100%, but won't it make it harder for us to debug issues when the problem lies in the gem/rails?

**Comment by rdubya** [comment]
What database are you testing with?

File: lib/arjdbc/jdbc/adapter.rb [hunk 2]
```diff
@@ -250,12 +250,6 @@ def exec_query_raw(sql, name = 'SQL', binds = [], &block)
         end
       end
 
-      # @private
-      # @override
-      def select_rows(sql, name = nil, binds = [])
```
Comment (kares) [inline lib/arjdbc/jdbc/adapter.rb]: also, I do not understand the removal from the desc or comit message: "Fixed issue with User.exists? 1 call"
could you elaborate? are you trying to fit the base adapter (un-used within core) with the DB2 adapter as is?

File: lib/arjdbc/abstract/core.rb (hunk 1 above)
Comment (rsov) [reply lib/arjdbc/abstract/core.rb]: Actually the opposite. All that method does is figure out which line executed the AR call and logs it into rails sever log if `ActiveRecord::Base.verbose_query_logs`. See [source](https://github.com/rails/rails/blob/fc5dd0b85189811062c85520fd70de8389b55aeb/activerecord/lib/active_record/log_subscriber.rb#L102). It already removes AR gem from stack trace. 

Without the fix (same line every time)

```
User Load (7.0ms)  SELECT  * FROM USERS WHERE id = ? LIMIT ?  [["id", "1"], ["LIMIT", 1]]
  ↳ /home/ruslan/rails/rsov/activerecord-jdbc-adapter/lib/arjdbc/abstract/core.rb:72
```

Expected (line changes based where it was called from)
```
User Load (7.0ms)  SELECT  * FROM USERS WHERE id = ? LIMIT ?  [["id", "1"], ["LIMIT", 1]]
  ↳ app/controllers/application_controller.rb:107


Member Load (59.2ms)  SELECT * FROM MEMBERS WHERE id = ? LIMIT ?  [["id", 1], ["LIMIT", 1]]
  ↳ app/views/layouts/_details.html.erb:6
```

File: lib/arjdbc/jdbc/adapter.rb (hunk 2 above)
Comment (rsov) [reply lib/arjdbc/jdbc/adapter.rb]: This was a tricky one.

When I execute query `User.find(1)` it does what you expect, executes this query: 
`SELECT  * FROM USERS WHERE id = ? LIMIT ?  [["id", "1"], ["LIMIT", 1]]`

However just doing `User.exists?(1)` does this:
```
ActiveRecord::StatementInvalid: ActiveRecord::JDBCError: [SQL0313] Number of host variables not valid.: SELECT  1 AS one FROM USER WHERE id = ? LIMIT ?
```

It lost the binds variables (`[["id", "1"], ["LIMIT", 1]]`)

Originally I though, yeah, another DB2 bug. But DB2 does not have `select_rows`. And if you print the binds variables from `select_rows`, it's empty

However the binds do exists. The first variable (`sql`) is changed to `arel` which contains building query information. Then AR52 [select_rows](https://github.com/rails/rails/blob/fc5dd0b85189811062c85520fd70de8389b55aeb/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb#L80) calls  [select_all](https://github.com/rails/rails/blob/fc5dd0b85189811062c85520fd70de8389b55aeb/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb#L46) which parses out the binds from the arel query.

Removing `select_rows` from JDBC fixed the issue

Comment (dr-itz) [reply lib/arjdbc/jdbc/adapter.rb]: ah yes, the AREL 5.2 change where the bind values are in the AST and need to be extracted. Nice catch.

**Comment by rsov** [comment]
@rdubya Currently testing on DB2/AS400

File: lib/arjdbc/abstract/core.rb (hunk 1 above)
Comment (kares) [reply lib/arjdbc/abstract/core.rb]: nice, thanks @rsov ... think it all makes sense.
@rdubya if you do not have anything else I'm fine with merging - feel free to go forward.

Comment (rdubya) [reply lib/arjdbc/abstract/core.rb]: Can you update it to create a module that we prepend to the class that then calls `super`? That way we don't have to come back and update this if they ever change what they are filtering out

Comment (rsov) [reply lib/arjdbc/abstract/core.rb]: @rdubya Done

Comment (rdubya) [reply lib/arjdbc/abstract/core.rb]: Cool. Thanks

**Pull request merged**